include/inviwo/core/processors/poolprocessor.h

```diff
@@ -0,0 +1,500 @@
+     *
+     * \code{.cpp}
+     * const auto calc = [image = inport_.getData()]
+     *     ( pool::Stop stop, pool::Progress progress) -> std::shared_ptr<const Image> {
+     *     if (stop) return nullptr;
+     *     auto newImage = std::shared_ptr<Image>(image->clone());
+     *     progress(0.5f);
+     *     // Do some work with the image
+     *     return newImage;
+     * };
+     *
+     * dispatchOne(calc, [this](std::shared_ptr<const Image> result) {
+     *     outport_.setData(result);
+     *     newResults();  // Let the network know that the processor has new results on the
+     *                    // outport.
+     * });
+     * \endcode
+     */
+    template <typename Job, typename Done>
+    void dispatchOne(Job&& job, Done&& done);
+
+    /**
+     * Dispatch a vector of background jobs. The jobs will be executed in a background thread in
+     * the thread pool. It is important that the jobs captures its state by value, since it might
+     * outlive the processor. The jobs can take two optional parameters
+     *     * pool::Stop a stop token to periodically check if the job has been canceled.
+     *     * pool::Progress a callback to report progress of the calculation. The progress is
+     *       represented as a float in the intervall [0.0, 1.0].
+     * If the jobs takes a progress callback the processor will show a progress bar. The progress
+     * from all the jobs will automatically be merged and normalized
+     *
+     * The second functor 'done' is called with the results when the background jobs are finished.
+     * It will be executed on the main thread, and only if the processor is still valid and the jobs
+     * have not been stopped. Hence it is safe to refer to the processor in this functor.
+     *
+     * \code{.cpp}
+     * std::vector<std::function<std::shared_ptr<Mesh>(pool::Stop, pool::Progress progress)>> jobs;
+     * for (...) {
+     *     jobs.push_back([some state](pool::Stop stop, pool::Progress progress) {
+     *         if (stop) return nullptr;
+     *         // construct some new Mesh
+     *         progress(0.5f);
+     *         return newMesh;
+     *     });
+     * }
+     * dispatchMany(jobs, [this](std::vector<std::shared_ptr<Mesh>> results) {
+     *     outport_.setData(std::make_shared<std::vector<std::shared_ptr<Mesh>>>(results));
+     *     newResults();
+     * });
+     * \endcode
+     */
+    template <typename Job, typename Done>
+    void dispatchMany(std::vector<Job> jobs, Done&& done);
+
+    /**
+     * handleError is called on the main thread whenever there has be an error in a background
+     * calculation this will by default just log the error message, and clear any outports. Deriving
+     * classes can overload this to enable custom error handling if needed.
+     */
+    virtual void handleError();
+
+    /**
+     * Call newResults to let the network know we have new data on our outports.
```

Comment:
Would prefer onNewResults

Reply:
it is not a callback like that...

Reply:
Perhaps newResultsAvailable then?
I think that it is kind of a notify-function so for me onNewResults makes sense still. 
For me, newResults could also be a creation function

Reply:
Perhaps newResultsAvailable then?
I think that it is kind of a notify-function so for me onNewResults makes sense still. 
For me, newResults could also be a creation function